File: frontend/src/app/model/MedicineIntake.ts
export class MedicineIntake{
    public id!: number;
    public therapy!: Therapy;
    public quantity!: number;
    public matchesTherapy!: boolean;
    public dateTime!: Date;

    constructor(data?: any){
        if(data) {
            this.id = data.id;
            this.therapy = new Therapy(data.therapy);
            this.quantity = data.quantity;
            this.matchesTherapy = data.matchesTherapy;
            this.dateTime = new Date(data.dateTime);
        }
    }
}

export class Therapy{
    public id!: number;
    public medicine!: string;
    public dosesPerDay!: number;
    public quantity!: number;
    public notes!: string;
  patient: any;

    constructor(data: any){
        this.id = data.id;
        this.medicine = data.medicine;
        this.dosesPerDay = data.dosesPerDay;
        this.quantity = data.quantity;
        this.notes = data.notes;
    }
}

File: backend/src/main/java/it/univr/DiabetesLogger/model/MedicineIntake.java
package it.univr.DiabetesLogger.model;

import jakarta.persistence.*;

import java.time.LocalDateTime;

@Entity
@Table(name = "medicine_intakes")
public class MedicineIntake {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    @ManyToOne
    @JoinColumn(nullable = false)
    private Patient patient;

    @ManyToOne
    @JoinColumn(nullable = false)
    private Therapy therapy;

    @Column(nullable = false)
    private Double quantity;

    @Column(nullable = false)
    private Boolean matchesTherapy;

    @Column(nullable = false)
    private LocalDateTime dateTime;

    protected MedicineIntake(){}

    public MedicineIntake(Patient patient, Therapy therapy, Double quantity, Boolean matchesTherapy, LocalDateTime dateTime){
        this.patient = patient;
        this.therapy = therapy;
        this.quantity = quantity;
        this.matchesTherapy = matchesTherapy;
        this.dateTime = dateTime;
    }

    public Integer getId(){
        return id;
    }

    public Patient getPatient(){
        return patient;
    }

    public void setPatient(Patient patient){
        this.patient = patient;
    }

    public Therapy getTherapy(){
        return therapy;
    }

    public void setTherapy(Therapy therapy){
        this.therapy = therapy;
    }

    public Double getQuantity(){
        return quantity;
    }

    public void setQuantity(Double quantity){
        this.quantity = quantity;
    }

    public Boolean getMatchesTherapy(){
        return matchesTherapy;
    }

    public void setMatchesTherapy(Boolean matchesTherapy){
        this.matchesTherapy = matchesTherapy;
    }

    public LocalDateTime getDateTime() {
        return dateTime;
    }

    public void setDateTime(LocalDateTime dateTime) {
        this.dateTime = dateTime;
    }
}
